src/connector/src/parser/postgres.rs

```diff
@@ -106,304 +59,65 @@ fn postgres_cell_to_scalar_impl(
     // Issue #1. The null of enum list is not supported in Debezium. An enum list contains `NULL` will fallback to `NULL`.
     // Issue #2. In our parser, when there's inf, -inf, nan or invalid item in a list, the whole list will fallback null.
     match data_type {
-        DataType::Boolean => {
-            handle_data_type!(row, i, name, bool)
-        }
-        DataType::Int16 => {
-            handle_data_type!(row, i, name, i16)
-        }
-        DataType::Int32 => {
-            handle_data_type!(row, i, name, i32)
-        }
-        DataType::Int64 => {
-            handle_data_type!(row, i, name, i64)
-        }
-        DataType::Float32 => {
-            handle_data_type!(row, i, name, f32)
-        }
-        DataType::Float64 => {
-            handle_data_type!(row, i, name, f64)
-        }
-        DataType::Decimal => {
-            handle_data_type!(row, i, name, Decimal)
-        }
-        DataType::Int256 => {
-            // Currently in order to handle the decimal beyond RustDecimal,
-            // we use the PgNumeric type to convert the decimal to a string.
-            // Then we convert the string to Int256.
-            // Note: It's only used to map the numeric type in upstream Postgres to RisingWave's rw_int256.
```

Comment:
All these debezium behavior related comments in `postgres_cell_to_scalar_impl` shall be moved to `ScalarAdapter::into_scalar`, especially on branches where `return None` and `builder.append(None)`.

Reply:
Updated